x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -615,6 +615,60 @@ private void clustersAndIndices(String command, String indexString1, String inde
         );
     }
 
+    public void testValidFromIndexPattern() {
```

Comment:
This has some overlap with `testStringAsIndexPattern` in the same file.

I think this test has the following drawbacks:
- Only one kind of quoting (`"`, but there's also still `"""`).
- None of the index patterns actually require quoting. Looking at `testStringAsIndexPattern`, there's e.g. date math stuff that can make quoting required.